src/org/sosy_lab/java_smt/solvers/cvc5/CVC5FloatingPointFormulaManager.java

```diff
@@ -409,36 +410,16 @@ protected Term isNegative(Term pParam) {
 
   @Override
   protected Term fromIeeeBitvectorImpl(Term bitvector, FloatingPointType pTargetType) {
-    int mantissaSize = pTargetType.getMantissaSize();
-    int exponentSize = pTargetType.getExponentSize();
-    int size = pTargetType.getTotalSize();
-    assert size == mantissaSize + exponentSize + 1;
-
-    Op signExtract;
-    Op exponentExtract;
-    Op mantissaExtract;
     try {
-      signExtract = termManager.mkOp(Kind.BITVECTOR_EXTRACT, size - 1, size - 1);
-      exponentExtract = termManager.mkOp(Kind.BITVECTOR_EXTRACT, size - 2, mantissaSize);
-      mantissaExtract = termManager.mkOp(Kind.BITVECTOR_EXTRACT, mantissaSize - 1, 0);
-    } catch (CVC5ApiException e) {
-      throw new IllegalArgumentException(
-          "You tried creating a invalid bitvector extract in term " + bitvector + ".", e);
+      return termManager.mkTerm(
+          termManager.mkOp(
+              Kind.FLOATINGPOINT_TO_FP_FROM_IEEE_BV,
+              pTargetType.getExponentSize(),
+              pTargetType.getMantissaSizeWithSignBit()),
+          bitvector);
+    } catch (CVC5ApiException pE) {
+      throw new RuntimeException(pE);
```

Comment:
Also looks like unrelated changes.

Reply:
Honestly, i would just add this to this PR. The previous implementation was working fine, but this built-in function does it automatically correctly. Since the old implementation needed precise exponent/mantissa etc., this was easier.